modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -300,6 +392,136 @@ public void remove(String regName) {
 
         if (mreg != null)
             notifyListeners(mreg, metricRegRemoveLsnrs, log);
+
+        metaLock.writeLock().lock();
```

Comment:
I don't quite get the intention behind the lock. What is it for?
DistributedMetaStorage is already internally synchronized. Can you please explain? Thank you.

Reply:
It doesn't necessary for now, indeed.
Removed.